### PR TITLE
ddev: init at 1.22.1

### PR DESCRIPTION
**Pull request by star-szr**
## Description of changes

Add [DDEV](https://ddev.com/).

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/virtualization/ddev/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+buildGoModule rec {
+  pname = "ddev";
+  version = "1.22.1";
+
+  src = fetchFromGitHub {
+    owner = "ddev";
+    repo = "ddev";
+    rev = "v${version}";
+    sha256 = "sha256-3uDdp827noiY6P7VSBXM34MTe9+MCEkNtuRQ3eFoBnk=";
```
Comment (AndersonTorres) [inline pkgs/applications/virtualization/ddev/default.nix:11]:
```suggestion
    hash = "sha256-3uDdp827noiY6P7VSBXM34MTe9+MCEkNtuRQ3eFoBnk=";
```

File: pkgs/applications/virtualization/ddev/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  };
+
+  vendorSha256 = null;
```
Comment (AndersonTorres) [inline pkgs/applications/virtualization/ddev/default.nix:14]:
```suggestion
  vendorHash = null;
```

File: pkgs/applications/virtualization/ddev/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  ldflags = [
+    "-extldflags -static"
+    "-X github.com/ddev/ddev/pkg/versionconstants.DdevVersion=${version}"
+    "-X github.com/ddev/ddev/pkg/versionconstants.SegmentKey=${version}"
+  ];
+
+  doCheck = false;
```
Comment (AndersonTorres) [inline pkgs/applications/virtualization/ddev/default.nix:22]: Explain why (in a comment):
```suggestion
  # <reason why>
  doCheck = false;
```

File: pkgs/applications/virtualization/ddev/default.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    version = "v${version}";
+  };
+
+  meta = with lib; {
+    description = "Docker-based local PHP+Node.js web development environments";
+    homepage = "https://ddev.com/";
+    license = licenses.asl20;
+    platforms = platforms.unix;
+    maintainers = with maintainers; [ star-szr ];
+  };
```
Comment (AndersonTorres) [inline pkgs/applications/virtualization/ddev/default.nix:36]: Don't forget to add `meta.mainProgram`.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Review by drupol — APPROVED** [review]

**Review by AndersonTorres — APPROVED** [review]

**Pull request merged**